packages/twenty-chrome-extension/src/utils/apolloClient.ts

```diff
@@ -1,67 +1,110 @@
-import { ApolloClient, from, HttpLink, InMemoryCache } from '@apollo/client';
+import {
+  ApolloClient,
+  from,
+  fromPromise,
+  HttpLink,
+  InMemoryCache,
+} from '@apollo/client';
 import { onError } from '@apollo/client/link/error';
 
+import { renewToken } from '~/db/token.db';
+import { Tokens } from '~/db/types/auth.types';
 import { isDefined } from '~/utils/isDefined';
 
 const clearStore = () => {
   chrome.storage.local.remove('loginToken');
-
   chrome.storage.local.remove('accessToken');
-
   chrome.storage.local.remove('refreshToken');
-
   chrome.storage.local.set({ isAuthenticated: false });
 };
 
-const getApolloClient = async () => {
+const setStore = (tokens: Tokens) => {
+  chrome.storage.local.set({
+    loginToken: tokens.loginToken,
+  });
+  chrome.storage.local.set({
+    accessToken: tokens.accessToken,
+  });
+  chrome.storage.local.set({
+    refreshToken: tokens.refreshToken,
+  });
+};
+
+export const getServerUrl = async () => {
   const store = await chrome.storage.local.get();
   const serverUrl = `${
     isDefined(store.serverBaseUrl)
       ? store.serverBaseUrl
       : import.meta.env.VITE_SERVER_BASE_URL
   }/graphql`;
+  return serverUrl;
+};
 
-  const errorLink = onError(({ graphQLErrors, networkError }) => {
-    if (isDefined(graphQLErrors)) {
-      for (const graphQLError of graphQLErrors) {
-        if (graphQLError.message === 'Unauthorized') {
-          //TODO: replace this with renewToken mutation
-          clearStore();
-          return;
-        }
-        switch (graphQLError?.extensions?.code) {
-          case 'UNAUTHENTICATED': {
-            //TODO: replace this with renewToken mutation
-            clearStore();
-            break;
+const getAuthToken = async () => {
+  const store = await chrome.storage.local.get();
+  if (isDefined(store.accessToken)) return `Bearer ${store.accessToken.token}`;
+  else return '';
+};
+
+const getApolloClient = async () => {
+  const store = await chrome.storage.local.get();
+  const errorLink = onError(
+    ({ graphQLErrors, networkError, forward, operation }) => {
+      if (isDefined(graphQLErrors)) {
+        for (const graphQLError of graphQLErrors) {
+          if (graphQLError.message === 'Unauthorized') {
```

Comment:
There might be a problem here, because the server doesn't necessarily returns this exact error.

I'm looking into it, might be a problem of guard.